src/com.company/heap/BthSmallestPrimeFraction.java

```diff
@@ -0,0 +1,103 @@
+package heap;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.PriorityQueue;
+
+public class BthSmallestPrimeFraction {
+
+    private int[] solve(int[] arr, int b) {
+        PriorityQueue<int[]> priorityQueue = new PriorityQueue<>(new Comparator<int[]>() {
+            @Override
+            public int compare(int[] o1, int[] o2) {
+                return Double.compare((double) arr[o1[0]] / arr[o1[1]], (double) arr[o2[0]] / arr[o2[1]]);
+            }
+        });
+
+        for (int i = 0; i < arr.length - 1; i++) {
+            priorityQueue.add(new int[]{i, arr.length - 1});
+        }
+        while (b > 0 && !priorityQueue.isEmpty()) {
+            int[] polledElement = priorityQueue.poll();
+            priorityQueue.add(new int[]{polledElement[0], polledElement[1] - 1});
```

Comment:
There is some wrong in this logic. You should verify the denominator index crosses the numerator index before inserting the element in the heap.This may lead to array index out of bound exception.